project-generation/helpers.go

```diff
@@ -0,0 +1,238 @@
+package projectgeneration
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"github.com/ONSdigital/log.go/log"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// validateArguments will ensure that all user input via flags have been provided and if not prompt for them to be and
+// keep prompting until all input meets validation criteria
+func validateArguments(ctx context.Context, repositoryCreated bool, unvalidatedName, unvalidatedType, unvalidatedProjectLocation, unvalidatedGoVersion string) (string, string, string, string, error) {
+	var validatedAppName, validatedProjectType, validatedProjectLocation, validatedGoVersion string
+	var err error
+	validatedAppName, err = ValidateAppName(ctx, unvalidatedName)
+	validatedProjectType, err = ValidateProjectType(ctx, unvalidatedType)
+	if err != nil {
+		return "", "", "", "", err
+	}
+	offerPurge := !repositoryCreated
+	validatedProjectLocation, err = ValidateProjectLocation(ctx, unvalidatedProjectLocation, validatedAppName, offerPurge)
```

Comment:
Cool so I agree it can technically be done like this, however, I have seen project get massively burnt by this and would highly recommend avoiding when possible. This is prob the longest comment I have ever made on a PR but would like to share what I have unfortunately seen take place where this was the root cause of the issue even though it may seem very minor right now.

Ultimately this is currently using the inverse of the 'data' stored in the `respositoryCreated` for the argument named `offerPurge`. However it is important to note that the variable name relationship `offerPurge == !respositoryCreated` is semantically incorrect [1]

In the past this kind of thing has caused issues where the next viewer of the code will want to modify the 'data' that is sent to the `offerPurge` argument. They see that `respositoryCreated` is being directly sent to the function so they simply add a one-liner to manipulate the variable based on a second requirement e.g.
`repositoryCreated = someFunc(repositoryCreated)` which now modifies the value of `repositoryCreated` before it is sent as the`offerPurge` argument. At this point, its name is no longer a representation of the data the variable holds, a quick and easy feature for programmer 2 to add and many code reviewers won't see the harm of this after all the variable is only being used once and is being used immediately on the next line. The code works so programmer 2 walks away thinking job done, quick, easy and simple. Without realising the issue of decoupling the name of the variable to the data that it holds - even if it was just one line.

Programmer 3 comes along and needs to get the program to do something else in this function based on if a repository has been created. Let's call this function `doANewThing()` they see the perfectly named variable and assume the data the variable holds is what it says on the tin (why wouldn't it be). This then leaves defects in the program as `doANewThing()` is now being called some times when it shouldn't and sometimes when it should.

You would think programmer 3 or the next programmer, programmer 4 would find the root cause of this issue and fix it... sadly this is not always the case and instead, programmer 3 creates a hacky fix and creates a new function that sets the value of `respositoryCreated` based on what they think the name of the variable means. This adds more layers to the code base, duplication and breaks the work of programmer 2. Then programmer 4 comes along and well you can see the direction this is going.

Lessons I learnt from this was
- Semantics are important to the human reader and a variable should only be used to represent the exact data it holds and nothing else.
- For any long-lived project, explicit > implicit 

[1] `canWalk == !cannotWalk` semantically this statement is correct